book/src/development/emitting_lints.md

```diff
@@ -0,0 +1,217 @@
+# Emitting a lint
+
+Once we have [defined a lint](define_lints.md), written [UI tests](write_tests.md)
+and chosen [the lint pass](lint_passes.md) for the lint, we can begin the
+implementation of the lint logic so that we can emit it and gradually work
+towards a lint that behaves as expected.
+
+Note that we will not go into concrete implementation of a lint logic in this
+chapter. We will go into details in later chapters as well as in two examples
+of real Clippy lints.
+
+In this chapter, we'll see an example of an imaginary `LateLintPass` lint named `bar_expressions`, that checks for
+expresions named `bar`.
```

Comment:
I would go here for a generic `LINT_NAME`. Especially because none of what's written in this chapter cares about what this lint does. And the `bar_expressions` name is not even used further down below.